src/imp/exec/pam.c

```diff
@@ -0,0 +1,67 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include "pam.h"
+
+static pam_handle_t *pam_h = NULL;
+
+
+int pam_setup (char *user)
+{
+    struct pam_conv conv = {misc_conv, NULL};
+    int rc = 0;
+    if ((rc = pam_start (FLUX_SERVICE_PAM, user, &conv, &pam_h)) != PAM_SUCCESS) {
+        printf ("pam_start: %s", pam_strerror (NULL, rc));
+        goto fail1;
+    } else if ((rc = pam_set_item (pam_h, PAM_USER, user)) != PAM_SUCCESS) {
```

Comment:
I know this style is probably cribbed from Slurm, but since there's a `goto fail1;`, the `else if` is unnecessary here. Let's remove the `else if` and just have
```suggestion
        goto fail1;
    }
    if ((rc = pam_set_item (pam_h, PAM_USER, user)) != PAM_SUCCESS) {
```

Reply:
If the `goto fail1` and `goto fail3` are only used once each, is it acceptable to just place the code for those statements where they're used?

Reply:
Well, in this case that would cause some code duplication. Because in all cases you need to reset `pam_h` to `NULL`, all the `goto fail*` can reuse this same block. It is probably safer to keep the `goto` statements in this case, also, because in the future if another conditional is added, the future developer is less likely to forget some error cleanup on failure.